.github/workflows/pr-test.yml

```diff
@@ -0,0 +1,79 @@
+# Run tests against developer push
+
+# Runs jobs on linux and windows to maintain compatibility.
+# The main repo is checked out into a sub-directory because an optional
+# second repo may be checked out for testing.
+
+# Optional GitHub secrets
+
+# See ci-helper.test.ts for information on the following
+# SMTP_USER = 'gitgitgadget.cismtpuser=<email_userid>'
+# SMTP_PASS = 'gitgitgadget.cismtpuser=<email_password>'
+# SMTP_HOST = 'gitgitgadget.cismtpuser=<email_hostname>'
+# SMTP_OPTS = 'gitgitgadget.cismtpuser=<email_smtp_options>'
```

Comment:
Two questions:
1. should we prefix those with `GGG_`, too?
2. Does it really make sense to require the secrets to contain all that non-secret stuff? I am starting to regret suggesting the `GIT_CONFIG_PARAMETERS` route...

Reply:
> 1. should we prefix those with `GGG_`, too?

I was going to suggest renaming the others but I suppose keeping them in a namespace is better to avoid conflicts if future vars are added by GitHub.

> 2. Does it really make sense to require the secrets to contain all that non-secret stuff? 

They are mostly secrets.  The `SMTP_OPTS` and `GGG_REPOSITORY` may not need to be.  These values are unique to each person who wants the tests run.

> I am starting to regret suggesting the `GIT_CONFIG_PARAMETERS` route...

Would you prefer I revert to the two git config steps?